### ISSUE TITLE
/root/.profile hardlinked to /.profile; /root on separate fs breaks upgrades

### ISSUE THREAD

**Issue by ltning** [ghostbsd/issues#2]
### Version

23.04.23

### Explain the issue and what happens

When doing a system upgrade using the update manager, the package installation enters a loop, installing all packages over and over.
When manually running `pkg upgrade`, the upgrade fails during
 `os-generic-userland-base-23.05.12`
with the message
 `pkg: Fail to create hardlink: /root/.pkgtemp..profile.th4ihI8OTWXD <-> /.pkgtemp..profile.wiPMoDJf57UC:Cross-device link`

The reason is that /root (on my system) is a separate dataset outside the tree from which boot environments are created:
```
Filesystem            Size    Used   Avail Capacity  Mounted on
tank2/ROOT/initial    743G     77G    666G    10%    /
tank2/root            667G    1.1G    666G     0%    /root
```
This is to ensure that contents of root's home directory are not touched in case of rollbacks and similar.

It is suggested that no hardlinks between these directories are made, and that /root/ is left alone during upgrades.

### Explain how to reproduce the bug?

See above.

### What is the Expected behavior (if applicable)

Successful upgrade, and no hardlink created or updated in /root.

### Additional context (if applicable)

Discussed on IRC today, 2023-05-19.

**Comment by ericbsd** [comment]
tank2/root is an old layout that has been removed. This error should not happen anymore. 

I need to look deeper to see what else I can do.

**Comment by ericbsd** [comment]
I am planning to move to a FreeBSD package base. They may have a way to avoid that issue.

**Comment by ericbsd** [comment]
This should be fixed  with [#152](https://github.com/ghostbsd/issues/issues/152)
